src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -745,8 +745,14 @@ private DocumentHandle AddDocument(DebugSourceDocument document, Dictionary<Debu
             DocumentHandle documentHandle;
             DebugSourceInfo info = document.GetSourceInfo();
 
+            var name = document.Location;
+            if (_sourceDebugDocumentsAdded && this.Context.RebuildDataOpt is { } rebuildData)
+            {
+                name = rebuildData.GetNextNonSourceFileDocumentName();
```

Comment:
It looks like this works by reading the document table in the original PDB and simply forcing us to give the same document names in the same order as the original. Is that right?

Reply:
Correct. That is the tricky part about this change. There are two parts and they have very different behavior:

- Path normalization: this is called in parallel with no defined order and the answers don't have to be the correct names, they just have to meet a few invariants documented in the function 
- Adding documents: this is called serially and is guaranteed to be called in the same order as the original build.